tests/unit/client.py

```diff
@@ -0,0 +1,659 @@
+        with pytest.raises(Exception):
+            C.node.remove_nic('node-08', 'eth0')
+
+    def test_node_connect_network(self):
+        result = C.node.connect_network(
+                'node-01', 'eth0', 'net-01', 'vlan/native'
+                )
+        assert result is None
+
+    def test_node_start_console(self):
+        result = C.node.start_console('node-01')
+        assert result is None
+
+    def test_node_stop_console(self):
+        result = C.node.stop_console('node-01')
+        assert result is None
+
+
+# FIXME: I spent some time on this test. Looks like the pytest
+# framework kills the network server before it can detach network.
+# def test_node_detach_network(self):
+# C.node.connect_network('node-04', 'eth0', 'net-04', 'vlan/native')
+# result = C.node.detach_network('node-04', 'eth0', 'net-04')
+# assert result is None
+
+
+@pytest.mark.usefixtures("create_setup")
+class Test_project:
+    """ Tests project related client calls."""
+
+    def test_list_projects(self):
+        """ test for getting list of project """
+        result = C.project.list()
+        assert result == [u'proj-01', u'proj-02', u'proj-03']
+
+    def test_list_nodes_inproject(self):
+        """ test for getting list of nodes connected to a project. """
+        result01 = C.project.nodes_in('proj-01')
+        result02 = C.project.nodes_in('proj-02')
+        assert result01 == [u'node-01']
+        assert result02 == [u'node-02', u'node-04']
+
+    def test_list_networks_inproject(self):
+        """ test for getting list of networks connected to a project. """
+        result = C.project.networks_in('proj-01')
+        assert result == [u'net-01', u'net-02', u'net-03']
+
+    def test_project_create(self):
+        """ test for creating project. """
+        result = C.project.create('dummy-01')
+        assert result is None
+
+    def test_duplicate_project_create(self):
+        """ test for catching duplicate name while creating new project. """
+        C.project.create('dummy-02')
+        with pytest.raises(Exception):
+            C.project.create('dummy-02')
+
+    def test_project_delete(self):
+        """ test for deleting project. """
+        C.project.create('dummy-03')
+        result = C.project.delete('dummy-03')
+        assert result is None
+
+    def test_error_project_delete(self):
+        """ test to capture error condition in project delete. """
+        with pytest.raises(Exception):
+            C.project.delete('dummy-03')
+
+    def test_project_connect_node(self):
+        """ test for connecting node to project. """
+        C.project.create('proj-04')
+        result = C.project.connect('proj-04', 'node-06')
+        assert result is None
+
+    def test_project_connect_node_duplicate(self):
+        """ test for erronous reconnecting node to project. """
+        C.project.create('proj-05')
+        C.project.connect('proj-05', 'node-07')
+        with pytest.raises(Exception):
```

Comment:
By "same with the rest of them," I mean the other tests need to be changed in the same way, not that it should be changed to what the other tests are doing.